backend/assignments/1/src/main/java/org/sanskar/coin/Coin.java

```diff
@@ -0,0 +1,86 @@
+package org.sanskar.coin;
+import org.sanskar.json.JsonReader;
+
+import java.util.logging.Logger;
+public class Coin {
+    private static Logger LOGGER = Logger.getLogger(Coin.class.getName());
+
+    private String rank;
+    private String name;
+    private String symbol;
+    private String price;
+    private String circulatingSupply;
+    private long volume;
+
+    public Coin(String rank, String name, String symbol, String price, String circulatingSupply) {
+        this.rank = rank;
+        this.name = name;
+        this.symbol = symbol;
+        this.price = price;
+        this.circulatingSupply = circulatingSupply;
+        this.volume = 0;
+    }
+
+    public String getRank() {
+        return rank;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getSymbol() {
+        return symbol;
+    }
+
+    public String getPrice() {
+        return price;
+    }
+
+    public long getVolume() {
+        return volume;
+    }
+
+    public String getCirculatingSupply() {
+        return circulatingSupply;
+    }
+
+
+
+    public synchronized void increaseVolume(long quantity) {
+        double currentVolume = Double.parseDouble(circulatingSupply);
+        circulatingSupply = String.valueOf((long) (currentVolume + quantity));
+    }
```

Comment:
Instead you should have applied lock on coin. Now what if for BTC you blocked it but ETH coins were available, resulted into ETH buyers are also waiting